src/messageBroker.js

```diff
@@ -0,0 +1,41 @@
+const Pubsub = require('orbit-db-pubsub')
+const redis = require('redis')
+
+const createMessage = (heads, id) => JSON.stringify({ from: id, heads })
+const messageParse = (message) => JSON.parse(message)
+
+class MessageBroker extends Pubsub {
+  constructor (ipfs, id, instanceId, redisOpts, onMessageCallback) {
+    super(ipfs, id)
+    this._topics = {}
+    this.instanceId = instanceId
+    this.onMessageCallback = onMessageCallback
+    this.messageClientSub = redis.createClient(redisOpts)
+    this.messageClientPub = redis.createClient(redisOpts)
```

Comment:
Why is there a different one for Sub and Pub?

Reply:
just a nuance in redis client, where once it subscribes to something, it goes into 'subscriber' mode. Basically a client cant be both subscriber and publisher.